bowtie/_cli.py

```diff
@@ -37,6 +37,7 @@
     Dialect,
     Example,
     Implementation,
+    NoSuchContainer,
```

Comment:
I'm in transit still today and tomorrow so I'll give this a closer read as soon as I can, but

1) you're definitely on the right track
2) try to cleanly separate between which modules have to know about your new thing -- in other words, the connectable nonsense is meant precisely to make sure that a module like this one does not need to know *how* it is speaking to implementations.

More specifically -- defining a new exception (`NoSuchContainer`) is likely perfectly fine, but catching it / letting it bubble up all the way here isn't -- by the time this module deals with it it should likely just be dealing with `NoSuchImplementation`, and something should have converted the `NoSuchContainer` into that -- basically at this module's layer it should simply be dealing with an exception which means "the user tried to talk to some connectable thing and that thing doesn't exist, regardless of whether it's an image name, a container id, whatever" -- I *think* `NoSuchImplementation` is already suited to be that but if not we can tweak it slightly. But yeah try to get your nice error message shown but without making it bubble all the way up here.

Reply:
Yeah you are right, I get it now. I'll display the message using the existing generic Exception (```StartuFailed```) in case connection establishment to a container fails.